test/unit/search/fm_index_cursor/fm_index_cursor_collection_test_template.hpp

```diff
@@ -321,6 +321,45 @@ TYPED_TEST_P(fm_index_cursor_collection_test, lazy_locate)
     EXPECT_TRUE(std::ranges::equal(it.locate(), it.lazy_locate()));
 }
 
+TYPED_TEST_P(fm_index_cursor_collection_test, locate_char_string)
```

Comment:
I thought after restructuring the text you can add this test in a different way (by a new fixture)?